src/routes.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+
+import { Navigation, PageWrapper } from './components';
+import { Home, Favorites, Details, Search, NotFound } from './views';
+import { SearchFormWrapper, SearchInput, SearchButton } from './components/search-form';
+
+const onSearch = (e, history) => {
+  e.preventDefault();
+  const { value } = e.target[0];
+  history.push(`/search/${encodeURI(value)}`);
+};
+
+export const RouteWithTopNav = ({ component: Component, ...rest }) => (
+  <Route {...rest} render={props => (
+    <div>
+      <Navigation />
+      <SearchFormWrapper onSubmit={e => onSearch(e, props.history)}>
+        <SearchInput type="text" placeholder="Search all the GIFs" />
```

Comment:
I believe most of these components are shared between all routes. Since we only care about the matching layout in this case the `Component` the rest of the components could be outside of each route